src/test/scala/uk/gov/nationalarchives/FileProcessorTest.scala

```diff
@@ -319,65 +326,67 @@ class FileProcessorTest extends AnyFlatSpec with MockitoSugar with TableDrivenPr
-                ).flatten
-              )
-            val files = List(
-              BagitFileMetadataObject(fileId, Option(assetId), fileName, 1, treFileName, 1),
-              BagitFileMetadataObject(metadataId, Option(assetId), "", 2, "metadataFileName.txt", 2)
-            )
-            val expectedBagitMetadataObjects: List[BagitMetadataObject] = List(folder, asset) ++ files
-
-            val fileProcessor =
-              new FileProcessor("download", "upload", "ref", mock[DAS3Client[IO]], UUIDGenerator().uuidGenerator)
-            val fileInfo = FileInfo(fileId, 1, treFileName, "fileChecksum")
-            val metadataFileInfo = FileInfo(metadataId, 2, "metadataFileName.txt", "metadataChecksum")
-
-            val bagitMetadataObjects =
-              fileProcessor
-                .createBagitMetadataObjects(
-                  fileInfo,
-                  metadataFileInfo,
-                  parsedUri,
-                  potentialCite,
+  forAll(fileReferenceTable) { (potentialFileReference, bornDigitalRefIdFields) =>
+    forAll(citeTable) { (potentialCite, idFields) =>
+      forAll(treNameTable) { (treName, treFileName, expectedFolderTitle, expectedAssetTitle) =>
+        forAll(urlDepartmentAndSeriesTable) { (department, series, _, parsedUri, expectedFolderName, titleExpected) =>
+          val updatedIdFields =
+            if (potentialCite.isDefined && expectedFolderName == trimmedUri) idFields :+ IdField("URI", trimmedUri)
+            else idFields
+          "createBagitMetadataObjects" should s"generate the correct bagit Metadata with $potentialFileReference $expectedFolderTitle, $expectedAssetTitle and $updatedIdFields" +
+            s"for $department, $series, $parsedUri and TRE name $treName" in {
+              val fileId = UUID.randomUUID()
+              val metadataId = UUID.randomUUID()
+              val folderId = uuids.head
+              val assetId = uuids.last
+              val fileName = treFileName.split("\\.").dropRight(1).mkString(".")
+              val folderTitle = if (titleExpected) Option(expectedFolderTitle) else None
+              val folder =
+                BagitFolderMetadataObject(folderId, None, folderTitle, expectedFolderName, updatedIdFields)
+              val asset =
+                BagitAssetMetadataObject(
+                  assetId,
+                  Option(folderId),
+                  expectedAssetTitle,
+                  expectedAssetTitle,
+                  List(fileId),
+                  List(metadataId),
                   treName,
-                  potentialUri,
-                  reference,
-                  department,
-                  series
+                  List(
+                    Option(IdField("UpstreamSystemReference", reference)),
+                    potentialUri.map(uri => IdField("URI", uri)),
+                    potentialCite.map(cite => IdField("NeutralCitation", cite))
+                  ).flatten ++ bornDigitalRefIdFields
```

Comment:
Is there a particular reason why you did `++ bornDigitalRefIdFields` as opposed to adding `potentialFileReference.map(fileReference => IdField("BornDigitalRef", fileReference)` inside the `List`?

Reply:
I could have had `fileReferenceTable` as a `TableFor1` and done it as you suggested. The only reason I had a two column table is this lets us see the expected id fields in the table and then add the expected fields directly to the `BagitAssetMetadataObject` 
As I'm typing this though, that seems overcomplicated so I've made this a `TableFor1` and done it as you suggested.